### DISCUSSION TITLE
How is the ancestral state stored in sgkit?

### DISCUSSION THREAD

**Discussion by hyanwong** [sgkit-dev/sgkit#580] [Q&A]
Hi - off the back of some questions about ancestral state polarisation triggered in tsinfer (https://github.com/tskit-dev/tsinfer/discussions/517), I wondered how sgkit intends to store ancestral state information, and how it might be possible to interface that to tools such as `tsinfer` or Peter Keightley's `est-sfs` software?

**Comment by hammer** [comment]
@hyanwong thanks for the inquiry! We don't currently store ancestral state information in `sgkit`. If you have a use case in mind, we'd love to learn more about it so that we might incorporate it into our library.

**Reply by hyanwong** [reply]
Thanks for the reply @hammer. I definitely have a use case in mind: it's important for our tree sequence inference pipeline (and also for things like the unfolded SFS). I'm sure @jeromekelleher has a better grasp of how `sgkit` will interface with `tsinfer`, but this is certainly one of the pieces that needs to be considered. At the moment, we simply require the ancestral state to be the first specified allele (i.e. index 0 in the samples x sites matrix). But it's helpful to be able to repolarise the ancestral state too (hence the reference to `est-sfs` which is a method for inferring the ancestral state).

Let me know how much more I need to explain, or maybe we should wait until @jeromekelleher is back in action.

**Reply by hammer** [reply]
I think that’s enough information for now! I filed https://github.com/pystatgen/sgkit/issues/585 with an idea for an approach that might work and will get @jeromekelleher’s thoughts when he returns. Thanks again for bringing this requirement to our attention.
